debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSnapshotChangeEventSource.java

```diff
@@ -350,8 +350,10 @@ protected void readTableStructure(ChangeEventSourceContext sourceContext,
                 .collect(Collectors.groupingBy(TableId::catalog, LinkedHashMap::new, Collectors.toList()));
         final Set<String> databases = tablesToRead.keySet();
 
-        // Record default charset
-        addSchemaEvent(snapshotContext, "", connection.setStatementFor(connection.readMySqlCharsetSystemVariables()));
+        if (!snapshottingTask.isBlocking()) {
+            // Record default charset
+            addSchemaEvent(snapshotContext, "", connection.setStatementFor(connection.readMySqlCharsetSystemVariables()));
```

Comment:
@jpechane I noted that for Charset statement no database name is provided. Is it right?

Reply:
Yes, IIRC it does not depnd on database